### PR TITLE
Disable device side malloc

### PR DESCRIPTION
**Pull request by yxsamliu**
Currently device side malloc is not working and takes excessive
device memory.

Disable it for now until a working malloc is implemented.

Change-Id: I1ad908c1c53a83752383b4be96688a848642c699

### CODE REVIEW

**Comment by yxsamliu** [comment]
Since this is urgent, and the existing PR has issues. I will just open a new one.

File: include/hip/hcc_detail/hip_runtime.h [hunk 1]
```diff
@@ -305,11 +310,15 @@ static constexpr Coordinates<hip_impl::WorkitemId> threadIdx{};
...
 #if __HCC_OR_HIP_CLANG__
+#if __HIP_ENABLE_MALLOC__
 extern "C" __device__ void* __hip_malloc(size_t);
 extern "C" __device__ void* __hip_free(void* ptr);
-
 static inline __device__ void* malloc(size_t size) { return __hip_malloc(size); }
 static inline __device__ void* free(void* ptr) { return __hip_free(ptr); }
+#else
+static inline __device__ void* malloc(size_t size) { return nullptr; }
+static inline __device__ void* free(void* ptr) { return nullptr; }
```
Comment (scchan) [inline include/hip/hcc_detail/hip_runtime.h:320]: Perhaps having a compilation error would be better?

Comment (yxsamliu) [reply include/hip/hcc_detail/hip_runtime.h:320]: There may be cases they need these functions to compile.
They will fail at runtime anyway.

Comment (scchan) [reply include/hip/hcc_detail/hip_runtime.h:320]: It's much easier to fix a build error than to debug a runtime failure.

Comment (yxsamliu) [reply include/hip/hcc_detail/hip_runtime.h]: ok. removed

**Comment by yxsamliu** [comment]
> Should we call this **HIP_ENABLE_DEVICE_MALLOC** ?

good point. fixed.

**Pull request merged**